### ISSUE TITLE
Fails to watch/serve after cargo update

### ISSUE THREAD

**Issue by eboody** [leptos-rs/leptos#479]
## Expected Behavior
Expected to be able to run `cargo leptos watch` in a new project

## Actual Behavior
Getting a thread panicked error

## Steps to Reproduce the Problem

1. `cargo install cargo-leptos`
2. `cargo leptos new --git https://github.com/leptos-rs/start`
3. `cd [your project name]`
4. `cargo update`
5. `cargo leptos watch`

## Specifications

  - Version: 0.1.0
  - Platform: Linux
  - Subsystem: Arch
  - Stack trace: 
```
  RUST_BACKTRACE=1 cargo leptos watch
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
       Cargo finished cargo build --package=leptos_start --lib --target-dir=target/front --target=wasm32-unknown-unknown --no-default-features --features=hydrate
thread 'tokio-runtime-worker' panicked at 'index out of bounds: the len is 0 but the index is 0', /home/eran/.cargo/registry/src/github.com-1ecc6299db9ec823/wasm-bindgen-cli-support-0.2.83/src/descriptor.rs:208:15
stack backtrace:
   0: rust_begin_unwind
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panicking.rs:64:14
   2: core::panicking::panic_bounds_check
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panicking.rs:147:5
   3: wasm_bindgen_cli_support::descriptor::Descriptor::_decode
   4: wasm_bindgen_cli_support::descriptor::Function::decode
   5: wasm_bindgen_cli_support::descriptor::Descriptor::_decode
   6: wasm_bindgen_cli_support::descriptors::execute
   7: wasm_bindgen_cli_support::Bindgen::generate_output
   8: cargo_leptos::compile::front::front::{{closure}}::{{closure}}
   9: tokio::runtime::task::core::Core<T,S>::poll
  10: tokio::runtime::task::harness::Harness<T,S>::poll
  11: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
  12: tokio::runtime::scheduler::multi_thread::worker::Context::run
  13: tokio::macros::scoped_tls::ScopedKey<T>::set
  14: tokio::runtime::scheduler::multi_thread::worker::run
  15: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
  16: tokio::runtime::task::core::Core<T,S>::poll
  17: tokio::runtime::task::harness::Harness<T,S>::poll
  18: tokio::runtime::blocking::pool::Inner::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Error: task 17 panicked
```

And here is the output of `cargo update` that broke things for me:

```
 ╰─λ cargo update
    Updating crates.io index
    Updating git repository `https://github.com/leptos-rs/leptos`
    Removing bincode v1.3.3
    Removing doc-comment v0.3.3
    Updating encoding_rs v0.8.31 -> v0.8.32
    Updating js-sys v0.3.60 -> v0.3.61
    Updating leptos v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_actix v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_config v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_dom v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_macro v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_meta v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_reactive v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_router v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating leptos_server v0.1.3 (https://github.com/leptos-rs/leptos#e2b1365e) -> #e380097a
    Updating parking_lot_core v0.9.6 -> v0.9.7
    Updating proc-macro2 v1.0.50 -> v1.0.51
    Removing rmp v0.8.11
    Removing rmp-serde v1.1.1
    Updating serde_json v1.0.91 -> v1.0.92
    Updating tinyvec_macros v0.1.0 -> v0.1.1
    Removing uuid v1.2.2
    Updating wasm-bindgen v0.2.83 -> v0.2.84
    Updating wasm-bindgen-backend v0.2.83 -> v0.2.84
    Updating wasm-bindgen-futures v0.4.33 -> v0.4.34
    Updating wasm-bindgen-macro v0.2.83 -> v0.2.84
    Updating wasm-bindgen-macro-support v0.2.83 -> v0.2.84
    Updating wasm-bindgen-shared v0.2.83 -> v0.2.84
    Updating web-sys v0.3.60 -> v0.3.61
      Adding windows-sys v0.45.0
      Adding windows-targets v0.42.1
```

**Comment by eboody** [comment]
Ok after a bunch of trial and error I found out that it was because of some mismatch in wasm-bindgen package versions. I had tried various permutation of deleting cargo lock, force installing older versions, and re-building and nothing worked.

Strangely, I was getting an error from a file that didn't exist on my system after deleting it and I couldn't understand where the error was even coming from so I deleting my .cargo folder entirely and re-ran `cargo leptos watch` and were back in business! 

Should I close this issue? Im curious if it has anything to do with leptos
